Python/chapter03/p01_three_in_one/miguelHx.py

```diff
@@ -0,0 +1,420 @@
+"""
+Python version 3.7.0
+3.1 - Three in one
+Describe how you could use a single array to implement three stacks
+"""
+
+import copy
+import unittest
+from typing import List
+
+
+class StackNode(object):
+
+    def __init__(self, data):
+        self.data = data
+        self.next = None # next is a pointer to a StackNode object
+
+class MyStack(object):
+    """
+    Stack data structure implementation.
+    Uses LIFO (last-in first-out) ordering.
+    The most recent item added to the stack is
+    the first removed.  Traversal is top to bottom.
+    """
+
+    def __init__(self):
+        self.top = None # top is a pointer to StackNode object
+        self.size = 0
+        return
+    
+    def pop(self) -> int:
+        """
+        Removes the top item from the stack
+
+        Raises:
+            IndexError: raised when pop is attempted on empty stack
+
+        Returns:
+            int: The data at the top of the stack
+        """
+        if self.top is None:
+            raise IndexError('Stack is Empty.')
+        item = self.top.data
+        self.top = self.top.next
+        self.size -= 1
+        return item
+    
+    def push(self, item: int) -> None:
+        """
+        Adds an item to the top of the stack
+        Args:
+            item (int): data we want at the top of stack
+        """
+        t = StackNode(item)
+        t.next = self.top
+        self.top = t
+        self.size += 1
+
+    def peek(self) -> int:
+        """
+        Returns data at the top of the stack
+
+        Raises:
+            IndexError: [description]
+
+        Returns:
+            int: the value at the top of the stack
+        """
+        if self.top is None:
+            raise IndexError('Stack is Empty')
+        return self.top.data
+    
+    def is_empty(self) -> bool:
```

Comment:
maybe this can use [`__bool__`](https://docs.python.org/3/reference/datamodel.html#object.__bool__) so that instead of `if !my_list.is_empty():` one can use `if my_list:`.

Reply:
Updated f0767768f1ad7ccbcabf2aba6e260b358ae44d1e